app/Libraries/Search/ForumSearchRequestParams.php

```diff
@@ -17,10 +19,46 @@ public function __construct(array $request)
         $this->username = presence(trim($request['username'] ?? null));
         $this->forumId = get_int($request['forum_id'] ?? null);
         $this->topicId = get_int($request['topic_id'] ?? null);
+        $this->parseSort(get_string($request['sort'] ?? null));
     }
 
     public function isLoginRequired(): bool
     {
         return true;
     }
+
+    private function parseSort(?string $sortStr): void
+    {
+        if ($sortStr === null) {
+            return;
+        }
+
+        $options = explode('_', $sortStr);
+        $field = $options[0];
+        $order = $options[1] ?? null;
+
+        if (!in_array($order, ['asc', 'desc'], true)) {
+            $order = static::DEFAULT_SORT_ORDER;
+        }
+
+        $validSort = true;
```

Comment:
Default to relevance if not valid? Which is the default anyway without the tie-breaker 🤔